### ISSUE TITLE
generator loss unnecessary?

### ISSUE THREAD

**Issue by RohithKuditipudi** [yashk2810/DCGAN-Keras#1]
Thanks for the implementation! Quick question: why do you compile the generator alone as a model with a binary cross-entropy loss? This seems unnecessary since the generator is only being trained when running the "gan" (i.e. g+d) model. Also, why do you set discriminator.trainable = True/False at various points during training? From my understanding of Keras, once you compile a model the trainable parameter is fixed for that model. So it's always the case that discriminator.trainable = true for the discriminator model and discriminator.trainable = false for the gan model.
